File: mycollab-services/src/main/java/com/mycollab/module/file/service/EntityUploaderService.java
package com.mycollab.module.file.service;

import java.awt.image.BufferedImage;

/**
 * @author MyCollab Ltd
 * @since 5.2.12
 */
public interface EntityUploaderService {
    /**
     * @param logo
     * @param basePath
     * @param oldId
     * @param uploadedUser
     * @param sAccountId
     * @param preferSizes
     * @return
     */
    String upload(BufferedImage logo, String basePath, String oldId, String uploadedUser,
                  Integer sAccountId, Integer[] preferSizes);
}

File: mycollab-services/src/main/java/com/mycollab/module/file/service/EntityUploaderService.kt
package com.mycollab.module.file.service

import java.awt.image.BufferedImage

/**
 * @author MyCollab Ltd
 * @since 5.2.12
 */
interface EntityUploaderService {
    /**
     * @param logo
     * @param basePath
     * @param oldId
     * @param uploadedUser
     * @param sAccountId
     * @param preferSizes
     * @return
     */
    fun upload(logo: BufferedImage, basePath: String, oldId: String, uploadedUser: String,
               sAccountId: Int?, preferSizes: Array<Int>): String
}
